backend/graph/api/schema.resolvers.go

```diff
@@ -79,31 +79,45 @@ func (r *mutationRootResolver) SetEpisodeProgress(ctx context.Context, id string
 	episodeID := utils.AsInt(e.ID)
 	q := r.Queries.ProfileQueries(p.ID)
 	var episodeProgress *common.Progress
+	pl := r.ProfileLoaders(ctx).ProgressLoader
 	if progress == nil {
 		err = q.ClearProgress(ctx, episodeID)
 	} else {
-		dur := e.Duration
-		if duration != nil {
-			dur = *duration
+		episodeProgress, err = pl.Get(ctx, utils.AsInt(e.ID))
+		if err != nil {
+			return nil, err
 		}
-		var showID null.Int
-		if e.Season != nil {
-			s, err := r.QueryRoot().Season(ctx, e.Season.ID)
-			if err != nil {
-				return nil, err
+		if episodeProgress == nil {
```

Comment:
should probably refactor due to the nesting here but I'm pretty sure we'll revisit it again

Reply:
I don't see a obvious way to do it either, so I guess its fine for now and we can extract into a method or something if really needed.